ldap/servers/slapd/conntable.c

```diff
@@ -125,12 +125,17 @@ connection_table_new(int table_size)
 {
     Connection_Table *ct;
     int i = 0;
+    int ct_list = 0;
+    int free_idx = 0;
     ber_len_t maxbersize = config_get_maxbersize();
 
     ct = (Connection_Table *)slapi_ch_calloc(1, sizeof(Connection_Table));
     ct->size = table_size;
-    ct->c = (Connection *)slapi_ch_calloc(1, table_size * sizeof(Connection));
-    ct->fd = (struct POLL_STRUCT *)slapi_ch_calloc(1, table_size * sizeof(struct POLL_STRUCT));
+    ct->list_num = 2;
```

Comment:
For bonus marks, consider https://man7.org/linux/man-pages/man3/numa.3.html  numa_num_configured_nodes() to get how many numa regions we have. Alternately, numa_num_configured_cpus()  / 4.

Reply:
Hi @Firstyear 
Bonus marks are always welcome :)

I understand the benefit of a process being executed on a NUMA node local to its CPU, does the scheduler policy not try make sure this happens. But, I don't quiet get  how the number of CT lists can be determined by the number of NUMA nodes on a system.

Reply:
> Hi @Firstyear Bonus marks are always welcome :)
> 
> I understand the benefit of a process being executed on a NUMA node local to its CPU, does the scheduler policy not try make sure this happens. But, I don't quiet get how the number of CT lists can be determined by the number of NUMA nodes on a system.

Since you tested that 2 CT are optimum on a machine with a single numa node, IMHO we should just use twice the number of nodes

Reply:
Yes, but we also then need to ensure that we are spawning the threads on differnt nodes ....

Reply:
Good point but that would means splitting the worker thread pool too.
Another point is that in the test Ludwig had done years ago on Solaris using machine with 64 cores, performance started to decrease when having more than 4 CTs .
So I suspect we are trying to get too far.  Maybe we should just have 2 CT if less than 8 CPUS and 4 otherwise. 
And do not bother about numa  for now. Anyway I do not think we could seriously improve things on numa without thinking globally.
(And to be honest, my feeling on that point, is to try to let the kernel handles the optimizations (including affinity handling) as far as possible rather than having the application doing it)

Reply:
Soundsgood, and it looks like @jchapma has done this, so we can resolve this point :)